overlord/snapstate/aliasesv2.go

```diff
@@ -0,0 +1,105 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snapstate
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/overlord/snapstate/backend"
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+// AliasTargets carries the targets of an alias in the context of snap.
+// If Manual is set it is the target of an enabled manual alias.
+// Auto is set to the target for an automatic alias, enabled or
+// disabled depending on the aliases status of the whole snap.
+type AliasTargets struct {
+	Manual string `json:"manual"`
+	Auto   string `json:"auto"`
+}
+
+// Effective returns the target to use based on the aliasStatus of the whole snap, returns "" if the alias is disabled.
+func (at *AliasTargets) Effective(aliasesStatus string) string {
```

Comment:
Nice name and signature.  I suggest using an AliasStatus type as well to make it more clear.